src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -1132,14 +1132,22 @@ public override Binder VisitXmlNameAttribute(XmlNameAttributeSyntax parent)
             /// </summary>
             private Binder GetParameterNameAttributeValueBinder(MemberDeclarationSyntax memberSyntax, Binder nextBinder)
             {
-                BaseMethodDeclarationSyntax baseMethodDeclSyntax = memberSyntax as BaseMethodDeclarationSyntax;
-                if ((object)baseMethodDeclSyntax != null && baseMethodDeclSyntax.ParameterList.ParameterCount > 0)
+                if (memberSyntax is BaseMethodDeclarationSyntax { ParameterList: { ParameterCount: > 0 } } baseMethodDeclSyntax)
                 {
                     Binder outerBinder = VisitCore(memberSyntax.Parent);
                     MethodSymbol method = GetMethodSymbol(baseMethodDeclSyntax, outerBinder);
                     return new WithParametersBinder(method.Parameters, nextBinder);
                 }
 
+                if (memberSyntax is RecordDeclarationSyntax { ParameterList: { ParameterCount: > 0 } } recordDeclSyntax)
+                {
+                    Binder outerBinder = VisitCore(memberSyntax.Parent);
+                    SourceNamedTypeSymbol recordType = ((NamespaceOrTypeSymbol)outerBinder.ContainingMemberOrLambda).GetSourceTypeMember(recordDeclSyntax);
+                    var primaryConstructor = recordType.GetMembersUnordered().OfType<SynthesizedRecordConstructor>().SingleOrDefault();
+                    Debug.Assert(primaryConstructor is not null);
+                    return new WithParametersBinder(primaryConstructor.Parameters, nextBinder);
```

Comment:
>nextBinder [](start = 83, length = 10)

What is this binder? Is it the same kind as the one used when we return from the previous ```if```?

Reply:
It's the binder used to bring parameters in scope into xmldocs. Yes, it's the same that is used in the previous `if` (which handles the xmldoc binding on methods).

```
    /// Binder used to place the parameters of a method, property, indexer, or delegate
    /// in scope when binding &lt;param&gt; tags inside of XML documentation comments.
```

Reply:
> It's the binder used to bring parameters in scope into xmldocs. Yes, it's the same that is used in the previous if (which handles the xmldoc binding on methods).

This doesn't really answer my question. Could you stop on both code paths under debugger and examine what is ```nextBinder``` on each of them? What is the type of the binder, containing symbols, etc.

---
In reply to: [518262702](https://github.com/dotnet/roslyn/pull/49134#discussion_r518262702) [](ancestors = 518262702)

Reply:
Sorry, I'd misunderstood the question (didn't notice you were pointing at "nextBinder").
`nextBinder` is a `BinderWithContainingMemberOrLambda` for the global namespace.

```
BuckStopsHereBinder
└─BinderWithContainingMemberOrLambda
  └─containing symbol: <global namespace>
```

But in the previous `if`, we'd have `nextBinder` being the containing type `C`:
```
BuckStopsHereBinder
└─BinderWithContainingMemberOrLambda
  └─containing symbol: C
```

You raise a good question (implicitly): should we chain a binder for the type `C` before we chain a binder for parameter list?

Reply:
> You raise a good question (implicitly): should we chain a binder for the type C before we chain a binder for parameter list?

If the binder that we return is used to resolve something other than parameter references, then, I think, we should add some targeted tests to verify/observe what the ```nextBinder``` brings/doesn't bring in scope and whether the behavior matches our expectations.

---
In reply to: [518301145](https://github.com/dotnet/roslyn/pull/49134#discussion_r518301145) [](ancestors = 518301145)

Reply:
I'd like to follow up offline on: 

> I think, we should add some targeted tests to verify/observe what the nextBinder brings/doesn't bring in scope and whether the behavior matches our expectations.

---
In reply to: [517015109](https://github.com/dotnet/roslyn/pull/49134#discussion_r517015109) [](ancestors = 517015109)

Reply:
Did some further review of impact.
The change to `BinderFactory.GetParameterNameAttributeValueBinder` only affects the name property as it is only used in `BinderFactory.VisitXmlNameAttribute`. Only `<param>` and `<paramref>` tags use it.
The resulting binder is only used in `DocumentationCommentCompiler.BindName`. I added some tests for some of the cases in that method.

---
In reply to: [541433186](https://github.com/dotnet/roslyn/pull/49134#discussion_r541433186) [](ancestors = 541433186,517015109)